litert/vendors/qualcomm/core/transformation/mha_to_sha.cc

```diff
@@ -0,0 +1,374 @@
+  auto& matmul1_v_output = tensor_pool.CloneNativeTensorFrom(
+      matmulv_cache_output, matmul1_v_output_dim);
+  EmplaceOpWithIO(new_ops, ops[start_id + kMatMulV1Index],
+                  {slice1_output, std::nullopt}, {matmul1_v_output});
+  // Slice 2
+  // Create StridedSlice param ranges.
+  auto mha_slice2_param =
+      ops[start_id + kSlice2Index].GetTensorPararm(0).GetTensor();
+  auto mha_slice2_param_data = mha_slice2_param.GetStaticTensorData<int32_t>();
+  std::vector<int32_t> slice2_param(mha_slice2_param_data.value().begin(),
+                                    mha_slice2_param_data.value().end());
+  slice2_param[7] /= num_heads;
+  auto& slice2_param_tensor = tensor_pool.CreateStaticTensor(
+      mha_slice2_param.GetDataType(), mha_slice2_param.GetQuantParams(),
+      mha_slice2_param.GetDims(), mha_slice2_param.GetTensorBytes(),
+      slice2_param.data());
+  // Create StridedSlice op.
+  auto slice2_output_dims =
+      ops[start_id + kSlice2Index].GetOutputTensor(0).GetDims();
+  slice2_output_dims[2] /= num_heads;
+  auto& slice2_output = tensor_pool.CloneNativeTensorFrom(
+      ops[start_id + kSlice2Index].GetOutputTensor(0), slice2_output_dims);
+  EmplaceOpWithIO(new_ops, ops[start_id + kSlice2Index], {softmax_output},
+                  {slice2_output});
+  new_ops.back().ClearParams();
+  new_ops.back().AddTensorParam(QNN_OP_STRIDED_SLICE_PARAM_RANGES,
+                                slice2_param_tensor);
+  // MatMul 2
+  const auto& matmulv_slice_output =
+      ops[start_id + kMatMulV2Index].GetOutputTensor(0);
+  std::vector<uint32_t> matmul2_v_output_dim = matmulv_slice_output.GetDims();
+  matmul2_v_output_dim[2] = matmul2_v_output_dim[2] / num_heads;
+  auto& matmul2_v_output = tensor_pool.CloneNativeTensorFrom(
+      matmulv_slice_output, matmul2_v_output_dim);
+  EmplaceOpWithIO(new_ops, ops[start_id + kMatMulV2Index],
+                  {slice2_output, std::nullopt}, {matmul2_v_output});
+  // Add
+  auto& add_final_output = tensor_pool.CloneNativeTensorFrom(
+      ops[start_id + kAdd2Index].GetOutputTensor(0),
+      matmul1_v_output.GetDims());
+  EmplaceOpWithIO(new_ops, ops[start_id + kAdd2Index],
+                  {matmul1_v_output, matmul2_v_output}, {add_final_output});
+  return add_final_output;
+}
+
+std::vector<OpWrapper> TransformToSHA(std::vector<OpWrapper>& ops,
+                                      size_t start_id, TensorPool& tensor_pool,
+                                      const qnn::TensorWrapper& mha_input,
+                                      const qnn::TensorWrapper& mha_output,
+                                      const ::qnn::OpWrapper& first_mul,
+                                      size_t num_heads) {
+  std::vector<OpWrapper> new_ops;
+
+  // Split
+  const std::vector<uint32_t> split_axis_dim{1};
+  const std::array<int32_t, 1> split_axis_data{2};
+  auto& split_axis = tensor_pool.CreateStaticTensor(
+      QNN_DATATYPE_INT_32, {}, split_axis_dim,
```

Comment:
```suggestion
      QNN_DATATYPE_INT_32, {}, {1},
```
and remove `split_axis_dim`

Reply:
Thank you!